### PR TITLE
Implement the `hook` interceptor

### PR DESCRIPTION
**Pull request by SimFG**
issue: #19216
Signed-off-by: SimFG <bang.fu@zilliz.com>

### CODE REVIEW

File: internal/core/thirdparty/rocksdb/CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,7 @@ macro( build_rocksdb )
         message ( STATUS "Turning on fPIC while building embedded Milvus" )
         set( FPIC_ARG "-DCMAKE_POSITION_INDEPENDENT_CODE=ON" )
     endif()
+    set( FPIC_ARG "-DCMAKE_POSITION_INDEPENDENT_CODE=ON" )
```
Comment (jennyli-z) [inline internal/core/thirdparty/rocksdb/CMakeLists.txt:30]: better remove if block instead of add the same line twice

**Comment by SimFG** [comment]
link: https://github.com/pingcap/tiflow/pull/834
https://github.com/BrainCoTech/libuv-prebuild/issues/1

**Comment by SimFG** [comment]
rerun ut

**Pull request merged**